frontend/test/metabase/scenarios/visualizations/pivot_tables.cy.spec.js

```diff
@@ -0,0 +1,75 @@
+import {
+  signInAsAdmin,
+  restore,
+  openOrdersTable,
+  popover,
+} from "__support__/cypress";
+
+describe("scenarios > visualizations > pivot tables", () => {
+  beforeEach(() => {
+    restore();
+    signInAsAdmin();
+  });
+
+  it("should be created from an ad-hoc question", () => {
+    cy.server();
+    cy.route("POST", "/api/dataset").as("dataset");
+    cy.route("POST", "/api/advanced_computation/pivot/dataset").as(
+      "pivotTableDataset",
+    );
+
+    openOrdersTable({ mode: "notebook" });
+    cy.findByText("Summarize").click();
+    cy.findByText("Count of rows").click();
+    cy.findByText("Pick a column to group by").click();
+    popover().within(() => {
+      cy.findByText(/Orders?$/).click(); // Collapse "Orders" to avoid additional scrolling
+      cy.findByText(/Users?$/).click();
+    });
+    cy.get(".ReactVirtualized__Grid").scrollTo("bottom"); // "Source" is not visible - we have to scroll
+    cy.findByText("Source").click();
+    // Add another metric
+    cy.get(".Icon-add")
+      .last() // This is fragile, but there's no other way to get to this element right now
+      .click();
+    popover().within(() => {
+      cy.findByText(/Products?$/).click();
+      cy.findByText("Category").click();
+    });
+    cy.findByText("Visualize").click();
```

Comment:
It'd be nice if we could skip this UI configuration and instead push the question we want in the URL. IIRC Tom tried that once and encountered issues loading the needed frontend code in Cypress. Maybe we could write a simple helper that takes the JSON question we want and base64 encodes it for the URL.

Reply:
(That's unrelated to this PR)

Reply:
That would be awesome, but there's an alternative for this case.
I can use API to:
1. store this question (with `bar` visualization),
2. Open it and use UI to change the visualization to pivot table (that would still make it "ad-hoc", right?)
3. Continue asserting on UI

Reply:
> that would still make it "ad-hoc", right?

Unfortunately, it wouldn't since the query is unchanged. You could add another breakout or change the aggregate to switch to ad-hoc, but that point it's probably simpler to just use the UI the whole way.

Reply:
See [this line](https://github.com/metabase/metabase/pull/14051/files#diff-d5b5ec136d796378e5808beb6b14135815b5dbcc1fd7b47a06c61b7ef042d5a7R12)

Reply:
I wasn't expected that to run as an ad-hoc query, but I see that it does!

I think that behavior could change in the future. We don't need to run that as ad-hoc since the display isn't part of the query. If we refine that logic to prioritize running saved cards, this test will no longer use an ad-hoc query.